Checkbox: Add a new parameter of small for the checkboxes

ritikamotwani opened the pull request:
Fixes: #197 

Adds a parameter of small for checkboxes to match the design: https://app.zeplin.io/project/5ba343bf0c399c33db483653/screen/5bc4b654103fa80994da6862

ritikamotwani committed:
Checkbox: Add a new parameter of small for the checkbox according to the design

rosslo commented:
@ritikamotwani you need to refer to abstract for this PR 😄  https://app.abstract.com/projects/da6f9bdf-2057-45bf-9dda-8dbf0d261eac/branches/master/commits/f2ad2a10e5ee2d5419830a7d68f2d30ce8995d56/files/DDF7C217-39F9-45D7-8C53-8D8C88D5FC37/layers/C5041BEC-528E-42F6-84BA-EB540AA09B90?mode=design

There is a new checkbox with border., would you mind implementing it within this PR?

rosslo commented on stories/Input/CheckboxStory.tsx:
```diff
@@ -22,6 +22,14 @@ const props = {
       description: 'Sets the label of checkbox.',
     },
+    {
+      name: 'small',
```
do you think using `size: "large" | "small" ` will be more scalable?

ritikamotwani replied:
Ooh if I use size and keep default one as small, then I won't have to change  both the projects. Thanks!

rosslo commented on stories/Input/CheckboxStory.tsx:
```diff
@@ -22,6 +22,14 @@ const props = {
+      name: 'small',
+      type: 'boolean',
+      defaultValue: 'false',
+      possibleValue: 'true|false',
```
I think we should have whitespace around `|` for better readability.

rosslo commented on src/Style/Input/CheckboxStyle.ts:
```diff
@@ -69,3 +69,7 @@ export const CheckboxContainer = styled.div`
   }
 `;
+
+interface CheckboxContainerProps {
```
Could we place `CheckboxContainerProps` at `Checkbox.tsx` to centralize the interface of component props?

Therefore, https://github.com/glints-dev/glints-aries/pull/267/files#diff-c80f91432b71eee8dd792beba827c3cdR46 can extend from `CheckboxContainerProps` to avoid redeclaring `small`.

ritikamotwani commented:
> @ritikamotwani you need to refer to abstract for this PR 😄 https://app.abstract.com/projects/da6f9bdf-2057-45bf-9dda-8dbf0d261eac/branches/master/commits/f2ad2a10e5ee2d5419830a7d68f2d30ce8995d56/files/DDF7C217-39F9-45D7-8C53-8D8C88D5FC37/layers/C5041BEC-528E-42F6-84BA-EB540AA09B90?mode=design
> 
> There is a new checkbox with border., would you mind implementing it within this PR?

Thanks @rosslo for the review. Didn't know this abstract 😛 
Yes, will do it in the same.

ritikamotwani commented:
@rosslo , the link isn't working for me for abstract. Do we need access to this? It shows a 404

rosslo commented:
@ritikamotwani oh I think so, tag you on related slack thread. 😄

ritikamotwani committed:
Checkbox: Add a prop size to the checkbox component

ritikamotwani committed:
Checkbox: Add a checkbox with borders

ritikamotwani commented:
@rosslo , this one is updated. Please take a look.
Thanks!

rosslo commented:
@ritikamotwani, the following are UI mismatch:
- checkbox icon size should be 18px x 18px
![image](https://user-images.githubusercontent.com/13249748/81040220-74664f00-8edd-11ea-9038-c2f7e24b390d.png)
- should have the `disabled` status
![image](https://user-images.githubusercontent.com/13249748/81040277-9e1f7600-8edd-11ea-9332-61ba4c2a5312.png)
- I think we shoud explicitly set `margin-right: 10px` to instead of `margin-right: 0.5em` for checkbox icon
![image](https://user-images.githubusercontent.com/13249748/81040536-269e1680-8ede-11ea-8073-83ee6801253b.png)

rosslo commented:
@ritikamotwani I think we should show the checkboxes with different props (e.g. border, disabled) on the storybook, so that the designer can do QA for us.

rosslo commented on src/Input/Checkbox/Checkbox.tsx:
```diff
@@ -22,17 +24,19 @@ const Checkbox: React.FunctionComponent<Props> = ({
   return (
     <CheckboxContainer
-      className={classNames('aries-checkbox', className)}
+      className={classNames('aries-checkbox', { border, checked }, className)}
```
I think we prefer prop to className. 🤔  https://glints.slack.com/archives/CNX1CEB7A/p1571367329035200

ritikamotwani replied:
Yes, yes. I had to remove them after testing the stylings. Sorry

ritikamotwani committed:
Checkbox: Use border as props and add the component to storybook

ritikamotwani commented:
> @ritikamotwani I think we should show the checkboxes with different props (e.g. border, disabled) on the storybook, so that the designer can do QA for us.

The disabled state also doesn't exist for the checkbox. I'll do it in a different PR. For now it's the checkbox, and the checkbox with the borders. Added it to the storybook

ritikamotwani commented:
@rosslo , this one is updated. 
For the font size, I have kept the 14px one as the default. Let me know if it's okay.

rosslo commented:
@ritikamotwani , I think there are two things need to be adjusted.
1. Since the padding area is hoverable, I think it should be clickable also.
2. We might need to remove the `outline` effect on focus.

![image](https://user-images.githubusercontent.com/13249748/81478422-efb66080-924f-11ea-8062-73b2641d75b1.png)

rosslo commented:
> > @ritikamotwani I think we should show the checkboxes with different props (e.g. border, disabled) on the storybook, so that the designer can do QA for us.
> 
> The disabled state also doesn't exist for the checkbox. I'll do it in a different PR. For now it's the checkbox, and the checkbox with the borders. Added it to the storybook

Could you open an issue for it in case we will forget it? 🙏

rosslo commented:
> @rosslo , this one is updated.
> For the font size, I have kept the 14px one as the default. Let me know if it's okay.

I think it is fine, just checked the 14px is the font-size of current checkbox on cdd site.

rosslo commented:
@ritikamotwani another suggestion is to display the `Checkbox` with `size` prop on storybook. 
I believe it will help the designers to do QC for us. 🙏

ritikamotwani commented:
> @ritikamotwani , I think there are two things need to be adjusted.
> 
> 1. Since the padding area is hoverable, I think it should be clickable also.
> 2. We might need to remove the `outline` effect on focus.
> 
> ![image](https://user-images.githubusercontent.com/13249748/81478422-efb66080-924f-11ea-8062-73b2641d75b1.png)

Ah, I had to ask this too. There's an outline explicitly set in the code,
```
&:focus > label {
    outline: 5px auto -webkit-focus-ring-color;
  }
```

No idea why it's set. It is causing the blue border thingy. I think I'll just remove it. I didn't see any use of it when we used the Checkbox in the project. 
Yes, makes sense, the box should be clickable 🙏

ritikamotwani committed:
Checkbox: Update the dtorybook with size and remove outline on focus

ritikamotwani commented:
@rosslo , I have removed the outline and updated the storybook.

But with the entire block getting clicked for the border button:
If I add the onClick property with the state change directly to the `CheckboxContainer` with an id, for the input I'll have to add checked=checked`. I can also use `stopPropagation` in the input's onClick
It works but the value of event.target changes to the label tag and event.target.value won't return the value passed anymore. And I believe the project is using that when checkbox has an onClick.

Though instead of the value it can now use the id.

This is how I am doing it.

```
<CheckboxContainer
      className={classNames('aries-checkbox', className)}
      role="checkbox"
      aria-labelledby={id}
      aria-checked={checked}
      tabIndex={0}
      size={size}
      border={border}
      checked={checked}
      onClick={handleClick}
      id={id}
    >
      <input
        type="checkbox"
        id={id}
        value={value}
        checked={checked}
        {...restProps}
      />
      <label htmlFor={id} tabIndex={-1} id={id}>
        {label || value}
      </label>
    </CheckboxContainer>
```

One option to solve this I can think of is, keeping the handleClick function in input, and somehow triggering the onClick of the `input` through the `CheckboxContainer's onClick`.

Updated it https://github.com/glints-dev/glints-aries/pull/267/commits/3183307dbe19a2ac00bbec1539f8af79ee6cfd0f
 Let me know what you think of this. event.target.value is accessible

ritikamotwani committed:
Checkbox: Make the entire box clickable for border with box

ritikamotwani committed:
Checkbox: Update the style for the checkbox square in the component

ritikamotwani commented:
Also just checked that the border radius for the checkbox squre is 2px, updated that too.

rosslo commented:
@ritikamotwani for the onClick issue, I think it can be resolved by adding the border style to `label` rather than`CheckboxContainer`.

ritikamotwani committed:
Checkbox: Add borders to label to avoid changing onClick

ritikamotwani commented:
Thanks @rosslo !
Yes, that was the best way! There's no change in functionality now. No breaking changes. I guess it's ready now. Would ask a designer to review after merge 🙏

ritikamotwani committed:
Icon: Add the file preview icon
